src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -273,6 +273,12 @@ private function autowireMethod(\ReflectionFunctionAbstract $reflectionMethod, a
 
                 // no default value? Then fail
                 if (!$parameter->isDefaultValueAvailable()) {
+                    // For core classes, isDefaultValueAvailable() can
+                    // be false when isOptional() returns true. If the
+                    // argument *is* optional, allow it to be missing
+                    if ($parameter->isOptional()) {
```

Comment:
@fabpot it was the default behavior, but @nicolas-grekas changed it to make the pass more predictible: https://github.com/weaverryan/symfony/commit/146f07471e8876da6e92c958897a5e798954e219#diff-62df969ae028c559d33ffd256de1ac49L261